### PR TITLE
Document initializing Turing

### PR DESCRIPTION
**Pull request by sethaxen**
This PR introduces a docs page showing how to initialize Turing with Pathfinder. Note that to initialize the metric, we'd need to work with AdvancedHMC directly, but this is really clunky right now due to the complexity of getting the unconstrained log density. Hopefully there's a simpler way than how I'm currently doing it!

### CODE REVIEW

File: docs/src/turing.md [hunk 1]
```diff
@@ -0,0 +1,100 @@
+# Using Pathfinder to initializing sampling with Turing
+
+If you have a _differentiable_ Turing model, then you can use Pathfinder to initialize its parameters.
```
Comment (storopoli) [inline docs/src/turing.md:3]: Just my two cents. _Differentiable_ Turing model means one that do not have for loops and integers? Like a model which would run in Zygote or ReverseDiff?

Maybe explain with a footnote?

Comment (sethaxen) [reply docs/src/turing.md:3]: Thanks, yeah I wrote the text hastily. Should say the technical requirements are that all parameters are continuous and the log-density can be differentiated by an AD framework.

Comment (storopoli) [reply docs/src/turing.md:3]: Yes, that what I got the intuition. But a footnote would solve this all. Thanks!

**Comment by sethaxen** [comment]
With #25, we can run Pathfinder on a Turing `DynamicPPL.Model` model as follows:
```julia
prob = optim_problem(model, MAP(); constrained=false)
pathfinder(prob, 1)
```
It's probably worth it to add a conditionally loaded interface for a Turing model that makes it easy to draw samples into a format useful for Turing users, e.g. as an `MCMCChains.Chains`.

**Pull request closed without merging**